hydromt/components/mesh.py

```diff
@@ -11,33 +11,370 @@
 import xugrid as xu
 from pyproj import CRS
 
-from hydromt import workflows
+from hydromt.components.base import ModelComponent
 from hydromt.gis.raster import GEO_MAP_COORD
-from hydromt.models.api import Model
+from hydromt.io.readers import read_nc
+from hydromt.models import Model
+from hydromt.workflows.mesh import (
+    create_mesh2d,
+    mesh2d_from_raster_reclass,
+    mesh2d_from_rasterdataset,
+    rename_mesh,
+)
 
-__all__ = ["MeshModel"]
```

Comment:
I think you still need the __all__ but his time with MeshModelComponent.
@savente93 previously xugrid was an optional package. How does this work now with components discovery? Can we still make this one discovered only if xugrid is installed? Or should in the end xugrid become a mandatory dependency of hydromt ?

Reply:
XUgrid will become a mandatory dependency in v1